doc/processes/release_process.md

```diff
@@ -0,0 +1,109 @@
+
+* Use a single code base to support the various JDK-levels.
+* Track the upstream OpenJDK releases.  
+* Use a consistent set of tags across OpenJ9 & extensions repos to 
+ensure that releases can be rebuilt or branched later.
+* OpenJ9 releases should be high quality and pass a common quality bar. 
+* Communicate the tags for a release to AdoptOpenJDK to ensure that
+binaries at the correct levels have been created. 
+* Clearly define the supported platforms
+* Should not regress performance from release to release.
+
+
+## Release cadence
+OpenJ9 should aim to release quarterly to match the OpenJDK long term 
+support (LTS) release cadence.  OpenJDK 8 is released quarterly and 
+other LTS releases will likely follow similar release cadences.  There 
+should be monthly milestone builds as the project builds towards the 
+next release.
+
+## Quality bar
+Work with the AdoptOpenJDK community to define a common quality bar for 
+any OpenJDK-based JDK.  The aim is to define a set of tests, both 
+functional and stress, and 3rd party applications that should pass for 
+every release.
+
+## Platform support
+Define the platforms that are supported by this release.  The set of 
+platforms will depend on the availability of hardware at OpenJ9 and 
+AdoptOpenJDK that can be used to validate the platform meets the quality 
+bar.  The code base may support more platforms than can be validated by 
+the quality.  Such platforms will not be officially included in the release.
+
+## Performance
+OpenJ9 should avoid regressing performance from release to release.
+This is harder to validate in the open project given the limitted 
+hardware resources available.  Work is being done at AdoptOpenJDK to 
+enable performance testing.
+
+# Release Process
+
+1. Create an issue to track the release and record the information required
+above, including the SHAs and tags.  This information will mostly be duplicated
+with the Eclipse release plan.
+1. Ask committers - via Slack and mailing list -  to refrain from committing any
+further changes until an "all clear" signal is broadcast.  Disable the promotion
+from the `openj9-omr` `master` branch to the `openj9` branch.
+1. Pick a candidate OpenJ9 level to mark as the initial release candidate.  This
+should be a level that has successfully passed the nightly testing builds at 
+AdoptOpenJDK.
+1. Tag the `openj9` & `openj9-omr` repos at the specified level with a tag of the
+following form: `openj9-#releaseNumber#RC#candidatenumber#`.  For the `0.8` 
+release, this would result in a `openj9-0.8RC1` tag.
+1. Tag the Extensions repos at the levels that passed the AdoptOpenJDK builds with
+the same tag.
+1. Sound the "all-clear". The OpenJ9 `master` branch and `openj9-omr` promotion 
+can be reopened for pull requests at this point.
+1. Rebuild the tagged levels and ensure they pass the quality bar defined by 
+AdoptOpenJDK.
+1. Provide a window of time (a week?) for any stakeholders to highlight any 
```

Comment:
I think a week is too short. I think we should be requesting feedback on milestone builds and splitting a release branch early.  Maybe the 2nd monthly milestone build? Even that represents only a month for everyone to run their tests, decide which ones are likely to be OpenJ9's "fault", to report them as issues with all relevant details, and then for us to diagnose and fix them.

I think we should branch early so that stakeholders have more time to report problems before we actually release it, without A) being sabotaged by any ongoing development work (which would otherwise encourage stakeholders to test *late* in our release cycle and encourage us not to fix our stakeholder's problems in the current release), or B) requiring everyone to step up in a very short period of time (which will encourage people not to test, will give us a false sense of quality, and ultimately drives users away).

I think the "early" release branch is also important so that the project can continue to operate even while a release is happening (something @andrewcraik mentioned). If someone wants to deliver something into the release in that last month, then we need to put extra vigiliance against it and the developer pays the cost to put it into two releases (git is pretty helpful here, actually). That additional delivery cost encourages developers to contribute features early rather than late (or to hold off to the next cycle), which is probably a good pressure from a project quality standpoint.

I think the "continuous" delivery model is an excellent aspiration for this project, and I support it as a direction. But I don't want our goals for continuous delivery to become a bar that blocks users from engaging with us because we set a high bar for their engagement. OpenJDK sets a lower bar here (I'm not throwing stones at all) in that they fork their repositories from the "mainline" pretty early which gives stakeholders a long time to soak and report problems.

If some stakeholders can afford to test us all the time, we welcome them with open arms, but let's not make it table stakes, at least not right out of the gate.